etc/kayobe/pulp.yml

```
@@ -26,6 +26,10 @@ pulp_enable_tls: false
 pulp_username: admin
 pulp_password: "{{ secrets_pulp_password }}"
 
+# Credentials for non-admin user within Pulp.
+pulp_stack_username: stack
+pulp_stack_password: "{{ secrets_pulp_stack_password | default('omit') }}"
```

Our "traditional" way of doing this is just to make the default an empty string i.e. `''`. I'd prefer to use that just for consistency.

---

The filter `default('omit')` has been used throughout SKC and would preferable to an empty string as this would mean users would only have to define `secrets_pulp_stack_password` in the `secrets.yml` and that is it. Whereas an empty string here would need to be assigned a value and then added to `secrets.yml` also this enforces consistency between deployments.

---

`default('omit') != default(omit)`
I might be missing something, but I don't see any usage of 'omit' (as a string) in SKC, Kayobe, or Kolla-Ansible

>would preferable to an empty string as this would mean users would only have to define secrets_pulp_stack_password in the secrets.yml and that is it. Whereas an empty string here would need to be assigned a value and then added to secrets.yml

I'm not sure I understand the difference. If we just update the conditions, wouldn't the behaviour be identical?
To be clear, my suggestion is to replace it with this:
```suggestion
pulp_stack_password: "{{ secrets_pulp_stack_password | default('') }}"
```
Though admittedly, I've checked and we use that less in SKC than I thought, it's Kayobe and Kolla-Ansible where we use the pattern mostly.